nixos/modules/services/continuous-integration/concourse/web.nix

```diff
@@ -0,0 +1,216 @@
+        Example content of environmentFile:
+        ```
+        CONCOURSE_POSTGRES_PASSWORD=********
+        ```
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services = {
+      concourse-web = {
+        description = "Concourse CI web";
+        after = [
+          "network.target"
+          "postgresql.target"
+        ];
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          User = cfg.user;
+          WorkingDirectory = "%S/concourse-web";
+          StateDirectory = "concourse-web";
+          StateDirectoryMode = "0700";
+          UMask = "0007";
+          ConfigurationDirectory = "concourse-web";
+          EnvironmentFile = cfg.environmentFile;
+          ExecStart = "${lib.getExe cfg.package} web ${builtins.concatStringsSep " " cfg.args}";
+          Restart = if cfg.autoRestart then "on-failure" else "no";
+          RestartSec = 15;
+          CapabilityBoundingSet = "";
+          # Security
+          NoNewPrivileges = true;
+          # Sandboxing
+          ProtectSystem = "strict";
+          ProtectHome = true;
+          PrivateTmp = true;
+          PrivateDevices = true;
+          PrivateUsers = true;
+          ProtectHostname = true;
+          ProtectClock = true;
+          ProtectKernelTunables = true;
+          ProtectKernelModules = true;
+          ProtectKernelLogs = true;
+          ProtectControlGroups = true;
+          RestrictAddressFamilies = [ "AF_UNIX AF_INET AF_INET6" ];
+          LockPersonality = true;
+          MemoryDenyWriteExecute = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          PrivateMounts = true;
+          # System Call Filtering
+          SystemCallArchitectures = "native";
+          SystemCallFilter = "~@clock @privileged @cpu-emulation @debug @keyring @module @mount @obsolete @raw-io @reboot @setuid @swap";
+        };
+        environment = {
+          CONCOURSE_BIND_PORT = toString cfg.network.bindPort;
+          CONCOURSE_PEER_ADDRESS = cfg.network.peerAddress;
+          CONCOURSE_EXTERNAL_URL = cfg.network.externalUrl;
+          CONCOURSE_API_MAX_CONNS = lib.mapNullable toString cfg.network.apiMaxConns;
+          CONCOURSE_BACKEND_MAX_CONNS = lib.mapNullable toString cfg.network.backendMaxConns;
+          CONCOURSE_CLUSTER_NAME = cfg.network.clusterName;
+
+          CONCOURSE_POSTGRES_PORT = lib.mapNullable toString cfg.postgres.port;
+          CONCOURSE_POSTGRES_SOCKET = cfg.postgres.socket;
+          CONCOURSE_POSTGRES_DATABASE = cfg.postgres.database;
+          CONCOURSE_POSTGRES_USER = cfg.postgres.user;
+          CONCOURSE_POSTGRES_PASSWORD = cfg.postgres.password;
+          CONCOURSE_SESSION_SIGNING_KEY = cfg.sessionSigningKey;
+          CONCOURSE_TSA_BIND_IP = cfg.tsa.bindIP;
+          CONCOURSE_TSA_BIND_PORT = toString cfg.tsa.bindPort;
+          CONCOURSE_TSA_HOST_KEY = cfg.tsa.hostKey;
+          CONCOURSE_TSA_AUTHORIZED_KEYS = cfg.tsa.authorizedKeys;
```

Comment:
In the spirit of [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md), I'd suggest that you simply define `environment` as a freeform modules and pass environment variables directly.
An example is https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/development/corteza.nix.